test/Microsoft.AspNetCore.Routing.Tests/Matchers/MatcherConformanceTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    public abstract class MatcherConformanceTest
+    {
+        internal abstract Matcher CreateMatcher(MatcherEndpoint endpoint);
+
+        [Fact]
+        public virtual async Task Match_SingleLiteralSegment_Success()
+        {
+            // Arrange
+            var (matcher, endpoint) = CreateMatcher("/simple");
+            var (httpContext, feature) = CreateContext("/simple");
+
+            // Act
+            await matcher.MatchAsync(httpContext, feature);
+
+            // Assert
+            DispatcherAssert.AssertMatch(feature, endpoint);
+        }
+
+        [Theory]
+        [InlineData("simple")]
+        [InlineData("/simple")]
+        [InlineData("~/simple")]
+        public virtual async Task Match_Sanitizies_TemplatePrefix(string template)
+        {
+            // Arrange
+            var (matcher, endpoint) = CreateMatcher(template);
+            var (httpContext, feature) = CreateContext("/simple");
+
+            // Act
+            await matcher.MatchAsync(httpContext, feature);
+
+            // Assert
+            DispatcherAssert.AssertMatch(feature, endpoint);
+        }
+
+        private static (HttpContext httpContext, IEndpointFeature feature) CreateContext(string path)
```

Comment:
I think we're going to use these helper methods all throughout the unit tests. It would be worth having them public on a test helper class of some kind.

Reply:
Well the idea is that all of the tests for matchers would subclass this. So I suppose it wouldn't be private.